spring-integration-core/src/main/java/org/springframework/integration/support/MutableMessageHeaders.java

```diff
@@ -0,0 +1,61 @@
+package org.springframework.integration.support;
+
+import org.springframework.messaging.MessageHeaders;
+
+import java.util.Map;
+import java.util.UUID;
+
+
+/**
+ * A MessageHeaders that permits direct access to and modification of the
+ * header map.
+ *
+ * @author Stuart Williams
+ * @since 4.2
+ */
+class MutableMessageHeaders extends MessageHeaders {
+
+    private static final long serialVersionUID = 3084692953798643018L;
+
+    /**
+     * @param headers map
+     */
+    public MutableMessageHeaders(Map<String, Object> headers) {
+        super(headers);
+    }
+
+    /**
+     * @param headers map
+     * @param id of message
+     * @param timestamp of message
+     */
+    public MutableMessageHeaders(Map<String, Object> headers, UUID id, Long timestamp) {
+        super(headers, id, timestamp);
+    }
+
+    @Override
+    public Map<String, Object> getRawHeaders() {
+        return super.getRawHeaders();
+    }
```

Comment:
Right. And my intention was about that fact that this `MutableMessageHeaders` is `package protected`. Hence we won't have any access to this `getRawHeaders()` as `public` method. From other side this method is needed here anyway (as `protected`) to get access from `MutableMessage`